js/scripts.js

```diff
@@ -1 +1,106 @@
-// js file
\ No newline at end of file
+var newGameElem = document.getElementById('js-newGameElement'),
+    newGameBtn = document.getElementById('js-newGameButton'),
+    pickElem = document.getElementById('js-playerPickElement'),
+    pickRock = document.getElementById('js-playerPick_rock'),
+    pickPaper = document.getElementById('js-playerPick_paper'),
+    pickScissors = document.getElementById('js-playerPick_scissors'),
+    resultsElem = document.getElementById('js-resultsTableElement'),
+    playerPointsElem = document.getElementById('js-playerPoints'),
+    playerNameElem = document.getElementById('js-playerName'),
+    computerPointsElem = document.getElementById('js-computerPoints'),
+    playerPickElem = document.getElementById('js-playerPick'),
+    computerPickElem = document.getElementById('js-computerPick'),
+    playerResultElem = document.getElementById('js-playerResult'),
+    computerResultElem = document.getElementById('js-computerResult'),
+    gameState = 'notStarted', //started // ended
+    player = {
+        name: '',
+        score: 0
+    },
+    computer = {
+        score: 0
+    };
+
+newGameBtn.addEventListener('click', newGame);
+
+pickRock.addEventListener('click', function() {
+    playerPick('rock')
+});
+pickPaper.addEventListener('click', function() {
+    playerPick('paper')
+});
+pickScissors.addEventListener('click', function() {
+    playerPick('scissors')
+});
+
+function setGameElements() {
+    switch (gameState) {
+        case 'started':
+            newGameElem.style.display = 'none';
+            pickElem.style.display = 'block';
+            resultsElem.style.display = 'block';
+            break;
+        case 'ended':
+            newGameBtn.innerText = 'Jeszcze raz';
+        case 'notStarted':
+        default:
+            newGameElem.style.display = 'block';
+            pickElem.style.display = 'none';
+            resultsElem.style.display = 'none';
+    }
+}
+
+function newGame() {
+    player.name = prompt('Please enter your name', 'imię gracza');
+    if (player.name) {
+        player.score = computer.score = 0;
+        gameState = 'started';
+        setGameElements();
```

Comment:
to samo się tyczy `player.name` - nie potrzebujesz globalnej zmiennej do trzymania imienia (póki co przynajmniej)